src/components/Interactive.tsx

```diff
@@ -1,16 +1,29 @@
 import React, { useState, useLayoutEffect, useRef, useCallback } from "react";
+
 import styles from "../styles.css";
 
 // Limit number within [0, 1] bounds.
 // Use ternary operator instead of `Math.min(Math.max(0, number), 1)` to save few bytes
-const limit = (number) => (number > 1 ? 1 : number < 0 ? 0 : number);
+const limit = (number: number) => (number > 1 ? 1 : number < 0 ? 0 : number);
+
+interface Interaction {
+  left: number;
+  top: number;
+}
 
-const Interactive = ({ onMove, children }) => {
-  const container = useRef();
+interface Props {
+  onMove: (interaction: Interaction) => void;
+  children: React.ReactNode;
+}
+
+const Interactive = ({ onMove, children }: Props) => {
```

Comment:
I'm not good at it, but should we use React.FC type everywhere? I mean:
```ts
const Interactive: React.FC<Props> = ({ onMove, children }) => (
```

See: https://github.com/typescript-cheatsheets/react#section-2-getting-started

Reply:
I've been using Preact heavily for the last 6 months or so and haven't touched React since, so I apologize, some of my typings might be a bit awkward.

Reply:
The type is inferred. It's not necessary, even in strict mode. Could type it, but it's essentially like doing:

```
const x: string = "hello world";
```

In fact, there's apparently [reasons to avoid `React.FC` altogether.](https://github.com/facebook/create-react-app/pull/8177)

Sorry, I've been playing around in Preact land for the past 6 months and it has entirely replaced React for me. Few differences in typing the two, so I might be out of touch with React specific typings, and now there seems to be a movement against `React.FC` as of recently.

Reply:
Alright.
I'm actually okay with any solution you choose here)